packages/core/src/scenes/Variables.ts

```diff
@@ -0,0 +1,49 @@
+import type {Scene} from './Scene';
+import {createSignal, SimpleSignal} from '../signals';
+
+export class Variables {
+  private signals: {[key: string]: SimpleSignal<any>} = {};
+
+  public constructor(private readonly scene: Scene) {
+    scene.onReset.subscribe(this.handleReset);
+  }
+
+  /**
+   * Get variable signal if exists or create signal if not
+   *
+   * @param name - The name of the variable.
+   */
+  public get<T>(name: string, initial: T): SimpleSignal<T> | null {
```

Comment:
One last tiny issue, I believe it's no longer possible for this method to return null:
```suggestion
  public get<T>(name: string, initial: T): SimpleSignal<T> {
```